bin/live/pycbc_live_collate_triggers

```diff
@@ -0,0 +1,306 @@
+    assert args.trigger_dir, 'Please provide a directory containing trigger files.'
+    # Convert dates to datetime to get the days array
+    start_date = datetime.strptime(args.start_date, '%Y-%m-%d').date()
+    end_date = datetime.strptime(args.end_date, '%Y-%m-%d').date()
+
+    # Create the days array
+    days = []
+    delta = timedelta(days=1)
+
+    # Loop through the days and add each date to the list
+    current_date = start_date
+    while current_date <= end_date:
+        days.append(current_date)
+        current_date += delta
+
+    # Convert to strings in a list
+    days = [str(date).replace('-', '_') for date in days]
+    num_days = str(len(days))
+
+    trigger_files = [
+        os.path.join(args.trigger_dir, day, trigger_file)
+        for day in days
+        for trigger_file in os.listdir(os.path.join(args.trigger_dir, day))
+    ]
+
+if args.trigger_file_method == 'start-num-days':
+    assert args.start_date, 'Please provide a start date.'
+    assert args.num_days, 'Please provide a number of days.'
+    assert args.trigger_dir, 'Please provide a directory containing trigger files.'
+    # Convert dates to datetime to get the days array
+    start_date = datetime.strptime(args.start_date, '%Y-%m-%d').date()
+    num_days = timedelta(days=args.num_days - 1)
+    end_date = start_date + num_days
+
+    # Create the days array
+    days = []
+    delta = timedelta(days=1)
+
+    # Loop through the days and add each date to the list
+    current_date = start_date
+    while current_date <= end_date:
+        days.append(current_date)
+        current_date += delta
+
+    # Convert to strings in a list
+    days = [str(date).replace('-', '_') for date in days]
+    num_days = str(len(days))
+
+    trigger_files = [
+        os.path.join(args.trigger_dir, day, trigger_file)
+        for day in days
+        for trigger_file in os.listdir(os.path.join(args.trigger_dir, day))
+    ]
+
+if args.trigger_file_method == 'gps-start-end-time':
+    assert args.gps_start_time, 'Please provide a GPS start time.'
+    assert args.gps_end_time, 'Please provide a GPS end time.'
+    assert args.trigger_dir, 'Please provide a directory containing trigger files.'
+
+    trigger_files = [
+        l for l in glob.glob(args.trigger_dir + '/*/*-Live-*.hdf', recursive=True)
+        if float(l.split('/')[-1].split("-")[2]) > args.gps_start_time and float(l.split('/')[-1].split("-")[2]) < args.gps_end_time
+    ]
+
+start = timeit.default_timer()
+logging.info(f" {len(trigger_files)} files found")
+
+if args.output_trigger_file_list:
+    logging.info(' Writing list of trigger files to: ' + args.output_trigger_file_list)
+    with open(args.output_trigger_file_list, 'w') as f:
+        for item in trigger_files:
+            f.write("%s\n" % item)
+
+# Record start and end gpstime for the output file
+#  This assumes the trigger files are sorted chronologically
+start_gpstime = float(trigger_files[0].split('/')[-1].split("-")[2])
```

Comment:
Please add a fixme that the trigger files should have the gps start and end times of the file added as attributes, and that the start/end times should be obtained by getting the min/max of these over the file reading loop

Reply:
we should not assume that the trigger files are sorted chronologically

Reply:
I stopped being lazy and extracted the gpstimes from the filenames and took the min/max